src/gui/src/mainwindow.cpp

```diff
@@ -0,0 +1,2197 @@
+/*
+ * This file is part of the QPxTool project.
+ * Copyright (C) 2007-2010 Gennady "ShultZ" Kozlov <qpxtool@mail.ru>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ * See the file "COPYING" for the exact licensing terms.
+ *
+ */
+
+#include <QFont>
+#include <QMenu>
+#include <QMenuBar>
+#include <QMessageBox>
+#include <QMimeData>
+#include <QPushButton>
+#include <QStatusBar>
+#include <QToolBar>
+#include <QToolButton>
+
+#include <QComboBox>
+#include <QFrame>
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QProgressBar>
+#include <QVBoxLayout>
+
+#include <QProcess>
+#include <QTcpSocket>
+
+#include <QSqlError>
+#include <QSqlQuery>
+
+#include <QBuffer>
+#include <QDateTime>
+#include <QFile>
+#include <QFileDialog>
+#include <QTextDocument>
+#include <QUrl>
+
+#include <QDragEnterEvent>
+#include <QDragLeaveEvent>
+#include <QDropEvent>
+
+#include <QApplication>
+
+#include "about.hpp"
+#include "colors.hpp"
+#include "db_report_selection.hpp"
+#include "defs.hpp"
+#include "device.hpp"
+#include "devsettings.hpp"
+#include "mainwidget.hpp"
+#include "preferences.hpp"
+#include "printpreview.hpp"
+#include "progresswidget.hpp"
+#include "qpxiodevice.hpp"
+#include "testdialog.hpp"
+
+#include "mainwindow.hpp"
+#include "version.hpp"
+
+#include <iostream>
+#include <unistd.h>
+
+#include <QDebug>
+
+#define PRINT_ERRC_DETAILED
+
+#define PRINT_GRAPH_SCALE 2.0
+
+#define HTML_GRAPH_W 600
+#define HTML_GRAPH_H 300
+#define HTML_GRAPH_WQ "600"
+#define HTML_GRAPH_HQ "300"
+#define HTML_GRAPH_HQ2 "150"
+
+const QString errcNameCD[8] = {"BLER", "E11", "E21", "E31",
+                               "E12",  "E22", "E32", "UNCR"};
+const QString errcNameDVD[8] = {"res", "PIE", "PI8", "PIF",
+                                "POE", "PO8", "POF", "UNCR"};
+const QString errcNameBD[8] = {"res", "LDC", "res", "res",
+                               "BIS", "res", "res", "UNCR"};
+
+#include <getopt.h>
+#include <typeinfo>
+
+static struct option long_options[] = {
+    {"load", 1, NULL, 'l'}, {"help", 0, NULL, 'h'}, {0, 0, 0, 0}};
+
+QPxToolMW::QPxToolMW(int ac, char **av, QWidget *p, Qt::WindowFlags fl)
+    : QMainWindow(p, fl) {
+  char c;
```

Comment:
Codacy found an issue: [The scope of the variable 'c' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)